### PR TITLE
Support text log

### PR DESCRIPTION
**Pull request by punitcse**
# What
- Support text logging in twiglet by converting text into a ECS format.
- Add few test to support that.
- some refactoring.

# Why
So we can save use the twiglet for libraries which are using text logging.

@simplybusiness/application-tooling

### CODE REVIEW

File: test/logger_test.rb [hunk 1]
```diff
@@ -12,191 +12,236 @@
...
-
-    expected_log = {
-      message: 'Out of pets exception',
-      "@timestamp": '2020-05-11T15:01:01.000Z',
-      service: {
-        name: 'petshop'
-      },
-      log: {
-        level: 'error'
+    it 'should log mandatory attributes' do
```
Comment (sleepyfox) [inline test/logger_test.rb:38]: I'm not seeing what the difference is between the before and after of the git diff here.

File: lib/twiglet/logger.rb [hunk 2]
```diff
@@ -65,12 +65,18 @@ def with(default_properties)
     private
 
     def log(level:, message:)
-      raise 'Message must be a Hash' unless message.is_a?(Hash)
-
-      message = message.transform_keys(&:to_sym)
-      message.key?(:message) || raise('Log object must have a \'message\' property')
-
-      message[:message].strip.empty? && raise('The \'message\' property of log object must not be empty')
+      case message
```
Comment (sleepyfox) [inline lib/twiglet/logger.rb:70]: Can I suggest that we subclass Logger here to have an ObjectLogger and a StringLogger, and for the ObjectLogger to be the default and to behave as it currently does, but the StringLogger to have the new functionality?

**Review by sleepyfox — CHANGES_REQUESTED** [review]

File: test/logger_test.rb (hunk 1 above)
Comment (punitcse) [reply test/logger_test.rb:38]: I have added a describe block before 'JSON Loggig' and `text logging` to separate the concerns

File: lib/twiglet/logger.rb (hunk 2 above)
Comment (punitcse) [reply lib/twiglet/logger.rb:70]: Sure, you suggested that yesterday I missed that. My bad.

I have few questions lets pair on this when you get some time.

File: lib/twiglet/logger.rb [hunk 3]
```diff
@@ -65,12 +65,18 @@ def with(default_properties)
     private
 
     def log(level:, message:)
```
Comment (moltenice) [inline lib/twiglet/logger.rb:69]: I think if we are making this change we should try and implement EVERY method interface from BOTH Rails and Ruby's internal logging but implement it to be ECS compliant to ensure most libraries trying to log will work with Twiglet.

**Comment by sleepyfox** [comment]
I disagree, just the ones that are actually used or we'll have a
wasteground of dead methods.

File: lib/twiglet/logger.rb (hunk 3 above)
Comment (punitcse) [reply lib/twiglet/logger.rb:69]: The only fear I have and as @sleepyfox said lot of unnecessary methods. 
I will try and see if we can use the ruby formatter in some ways, if I get some time today.

**Review by sleepyfox — APPROVED** [review]
:pear:

**Pull request merged**